### ISSUE TITLE
fix errors in sphinx documentation generation steps

### ISSUE THREAD

**Issue by prasadtalasila** [prasadtalasila/MailingListParser#68]
The `make html` step of [sphinx documentation generation tutorial](https://github.com/prasadtalasila/MailingListParser/wiki/Generating-documentation-using-Sphinx) is giving an error. The step works very well on travis.

We need to trouble shoot this error so that we can auto deploy the project on an on-campus server.

**Comment by yashpungaliya** [comment]
I found a solution for this...
Update your sphinx using sudo pip3 install --upgrade sphinx
And build the html files using "sudo make html" instead of "make html"
I have fixed this on the server  and docs have been generated.

**Comment by prasadtalasila** [comment]
@yashpungaliya You are right. The documentation is being generated on the server. Please update [sphinx generation page](https://github.com/prasadtalasila/MailingListParser/wiki/Generating-documentation-using-Sphinx). Just to keep you updated, the remedy is not working on my machine.

Once you update the wiki page, I will try again with the help of the tutorial. 

However, the log file generated during the execution of `make html` is as follows.
```shell
mailinglistparser@mailinglistparser:~/mailinglistparser/docs$ sudo make html
Running Sphinx v1.6.3
making output directory...
loading pickled environment... not yet created
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 22 source files that are out of date
updating environment: 22 added, 0 changed, 0 removed
reading sources... [100%] util                                                                                                                 
/home/mailinglistparser/mailinglistparser/docs/algos.rst:53: WARNING: Bullet list ends without a blank line; unexpected unindent.
WARNING: /home/mailinglistparser/mailinglistparser/docs/analysis.author.rst:17: (WARNING/2) autodoc: failed to import module 'analysis.author.curve_fitting'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/sphinx/ext/autodoc.py", line 657, in import_object
    __import__(self.modname)
  File "/home/mailinglistparser/mailinglistparser/lib/analysis/author/curve_fitting.py", line 3, in <module>
    from sklearn.metrics import mean_squared_error
ImportError: No module named 'sklearn'
WARNING: /home/mailinglistparser/mailinglistparser/docs/analysis.author.rst:25: (WARNING/2) autodoc: failed to import module 'analysis.author.edge_list'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/sphinx/ext/autodoc.py", line 657, in import_object
    __import__(self.modname)
  File "/home/mailinglistparser/mailinglistparser/lib/analysis/author/edge_list.py", line 7, in <module>
    from util.read_utils import *
  File "/home/mailinglistparser/mailinglistparser/lib/util/read_utils.py", line 6, in <module>
    import networkx as nx
ImportError: No module named 'networkx'
WARNING: /home/mailinglistparser/mailinglistparser/docs/analysis.author.rst:33: (WARNING/2) autodoc: failed to import module 'analysis.author.ranking'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/sphinx/ext/autodoc.py", line 657, in import_object
    __import__(self.modname)
  File "/home/mailinglistparser/mailinglistparser/lib/analysis/author/ranking.py", line 1, in <module>
    from util.read_utils import *
  File "/home/mailinglistparser/mailinglistparser/lib/util/read_utils.py", line 6, in <module>
    import networkx as nx
ImportError: No module named 'networkx'
WARNING: /home/mailinglistparser/mailinglistparser/docs/analysis.author.rst:41: (WARNING/2) autodoc: failed to import module 'analysis.author.time_statistics'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/sphinx/ext/autodoc.py", line 657, in import_object
    __import__(self.modname)
  File "/home/mailinglistparser/mailinglistparser/lib/analysis/author/time_statistics.py", line 1, in <module>
    from util.read_utils import *
  File "/home/mailinglistparser/mailinglistparser/lib/util/read_utils.py", line 6, in <module>
    import networkx as nx
ImportError: No module named 'networkx'
WARNING: /home/mailinglistparser/mailinglistparser/docs/analysis.author.rst:49: (WARNING/2) autodoc: failed to import module 'analysis.author.wh_table'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/sphinx/ext/autodoc.py", line 657, in import_object
    __import__(self.modname)
  File "/home/mailinglistparser/mailinglistparser/lib/analysis/author/wh_table.py", line 2, in <module>
    from util.read_utils import *
  File "/home/mailinglistparser/mailinglistparser/lib/util/read_utils.py", line 6, in <module>
    import networkx as nx
ImportError: No module named 'networkx'
WARNING: /home/mailinglistparser/mailinglistparser/docs/analysis.author.graph.rst:10: (WARNING/2) autodoc: failed to import module 'analysis.author.graph.author_interaction'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/sphinx/ext/autodoc.py", line 657, in import_object
    __import__(self.modname)
  File "/home/mailinglistparser/mailinglistparser/lib/analysis/author/graph/author_interaction.py", line 8, in <module>
    from util.read_utils import *
  File "/home/mailinglistparser/mailinglistparser/lib/util/read_utils.py", line 6, in <module>
    import networkx as nx
ImportError: No module named 'networkx'
WARNING: /home/mailinglistparser/mailinglistparser/docs/analysis.author.graph.rst:19: (WARNING/2) autodoc: failed to import module 'analysis.author.graph.bipartite'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/sphinx/ext/autodoc.py", line 657, in import_object
    __import__(self.modname)
  File "/home/mailinglistparser/mailinglistparser/lib/analysis/author/graph/bipartite.py", line 8, in <module>
    from util.read_utils import *
  File "/home/mailinglistparser/mailinglistparser/lib/util/read_utils.py", line 6, in <module>
    import networkx as nx
ImportError: No module named 'networkx'
WARNING: /home/mailinglistparser/mailinglistparser/docs/analysis.author.graph.rst:27: (WARNING/2) autodoc: failed to import module 'analysis.author.graph.generate'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/sphinx/ext/autodoc.py", line 657, in import_object
    __import__(self.modname)
  File "/home/mailinglistparser/mailinglistparser/lib/analysis/author/graph/generate.py", line 7, in <module>
    from util.read_utils import *
  File "/home/mailinglistparser/mailinglistparser/lib/util/read_utils.py", line 6, in <module>
    import networkx as nx
ImportError: No module named 'networkx'
WARNING: /home/mailinglistparser/mailinglistparser/docs/analysis.thread.rst:17: (WARNING/2) autodoc: failed to import module 'analysis.thread.hypergraph'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/sphinx/ext/autodoc.py", line 657, in import_object
    __import__(self.modname)
  File "/home/mailinglistparser/mailinglistparser/lib/analysis/thread/hypergraph.py", line 8, in <module>
    from util.read_utils import *
  File "/home/mailinglistparser/mailinglistparser/lib/util/read_utils.py", line 6, in <module>
    import networkx as nx
ImportError: No module named 'networkx'
WARNING: /home/mailinglistparser/mailinglistparser/docs/analysis.thread.rst:25: (WARNING/2) autodoc: failed to import module 'analysis.thread.message_activity'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/sphinx/ext/autodoc.py", line 657, in import_object
    __import__(self.modname)
  File "/home/mailinglistparser/mailinglistparser/lib/analysis/thread/message_activity.py", line 1, in <module>
    import plotly as ply
ImportError: No module named 'plotly'
WARNING: /home/mailinglistparser/mailinglistparser/docs/analysis.thread.rst:33: (WARNING/2) autodoc: failed to import module 'analysis.thread.ps_table'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/sphinx/ext/autodoc.py", line 657, in import_object
    __import__(self.modname)
  File "/home/mailinglistparser/mailinglistparser/lib/analysis/thread/ps_table.py", line 2, in <module>
    from util.read_utils import *
  File "/home/mailinglistparser/mailinglistparser/lib/util/read_utils.py", line 6, in <module>
    import networkx as nx
ImportError: No module named 'networkx'
WARNING: /home/mailinglistparser/mailinglistparser/docs/analysis.thread.rst:41: (WARNING/2) autodoc: failed to import module 'analysis.thread.time_statistics'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/sphinx/ext/autodoc.py", line 657, in import_object
    __import__(self.modname)
  File "/home/mailinglistparser/mailinglistparser/lib/analysis/thread/time_statistics.py", line 13, in <module>
    from util.read_utils import *
  File "/home/mailinglistparser/mailinglistparser/lib/util/read_utils.py", line 6, in <module>
    import networkx as nx
ImportError: No module named 'networkx'
WARNING: /home/mailinglistparser/mailinglistparser/docs/analysis.thread.rst:49: (WARNING/2) autodoc: failed to import module 'analysis.thread.wh_table'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/sphinx/ext/autodoc.py", line 657, in import_object
    __import__(self.modname)
  File "/home/mailinglistparser/mailinglistparser/lib/analysis/thread/wh_table.py", line 2, in <module>
    from util.read_utils import *
  File "/home/mailinglistparser/mailinglistparser/lib/util/read_utils.py", line 6, in <module>
    import networkx as nx
ImportError: No module named 'networkx'
WARNING: /home/mailinglistparser/mailinglistparser/docs/analysis.thread.graph.rst:10: (WARNING/2) autodoc: failed to import module 'analysis.thread.graph.edge_list'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/sphinx/ext/autodoc.py", line 657, in import_object
    __import__(self.modname)
  File "/home/mailinglistparser/mailinglistparser/lib/analysis/thread/graph/edge_list.py", line 2, in <module>
    from util.read_utils import lines_per_n
  File "/home/mailinglistparser/mailinglistparser/lib/util/read_utils.py", line 6, in <module>
    import networkx as nx
ImportError: No module named 'networkx'
WARNING: /home/mailinglistparser/mailinglistparser/docs/analysis.thread.graph.rst:18: (WARNING/2) autodoc: failed to import module 'analysis.thread.graph.generate'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/sphinx/ext/autodoc.py", line 657, in import_object
    __import__(self.modname)
  File "/home/mailinglistparser/mailinglistparser/lib/analysis/thread/graph/generate.py", line 7, in <module>
    import networkx as nx
ImportError: No module named 'networkx'
/home/mailinglistparser/mailinglistparser/docs/analysis_techniques.rst:47: WARNING: Inline strong start-string without end-string.
/home/mailinglistparser/mailinglistparser/docs/analysis_techniques.rst:204: WARNING: Enumerated list ends without a blank line; unexpected unindent.
WARNING: /home/mailinglistparser/mailinglistparser/docs/code_documentation.rst:4: (WARNING/2) autodoc: failed to import module 'lib'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/sphinx/ext/autodoc.py", line 657, in import_object
    __import__(self.modname)
ImportError: No module named 'lib'
WARNING: /home/mailinglistparser/mailinglistparser/docs/code_documentation.rst:10: (WARNING/2) autodoc: failed to import module 'lib.input'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/sphinx/ext/autodoc.py", line 657, in import_object
    __import__(self.modname)
ImportError: No module named 'lib'
WARNING: /home/mailinglistparser/mailinglistparser/docs/code_documentation.rst:17: (WARNING/2) autodoc: failed to import module 'lib.analysis.author'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/sphinx/ext/autodoc.py", line 657, in import_object
    __import__(self.modname)
ImportError: No module named 'lib'
WARNING: /home/mailinglistparser/mailinglistparser/docs/code_documentation.rst:24: (WARNING/2) autodoc: failed to import module 'lib.analysis.thread'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/sphinx/ext/autodoc.py", line 657, in import_object
    __import__(self.modname)
ImportError: No module named 'lib'
WARNING: /home/mailinglistparser/mailinglistparser/docs/code_documentation.rst:31: (WARNING/2) autodoc: failed to import module 'lib.util'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/sphinx/ext/autodoc.py", line 657, in import_object
    __import__(self.modname)
ImportError: No module named 'lib'
WARNING: /home/mailinglistparser/mailinglistparser/docs/code_documentation.rst:38: (WARNING/2) autodoc: failed to import module 'lib.deprecated'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/sphinx/ext/autodoc.py", line 657, in import_object
    __import__(self.modname)
ImportError: No module named 'lib'
WARNING: /home/mailinglistparser/mailinglistparser/docs/driver_author_analysis.rst:4: (WARNING/2) autodoc: failed to import module 'driver_author_analysis'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/sphinx/ext/autodoc.py", line 657, in import_object
    __import__(self.modname)
  File "/home/mailinglistparser/mailinglistparser/lib/driver_author_analysis.py", line 1, in <module>
    from analysis.author.curve_fitting import generate_crt_curve_fits
  File "/home/mailinglistparser/mailinglistparser/lib/analysis/author/curve_fitting.py", line 3, in <module>
    from sklearn.metrics import mean_squared_error
ImportError: No module named 'sklearn'
WARNING: /home/mailinglistparser/mailinglistparser/docs/driver_headers_imap.rst:4: (WARNING/2) autodoc: failed to import module 'driver_headers_imap'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/sphinx/ext/autodoc.py", line 657, in import_object
    __import__(self.modname)
  File "/home/mailinglistparser/mailinglistparser/lib/driver_headers_imap.py", line 1, in <module>
    from analysis.thread import generate_edge_list
ImportError: cannot import name 'generate_edge_list'
WARNING: /home/mailinglistparser/mailinglistparser/docs/driver_headers_mbox.rst:4: (WARNING/2) autodoc: failed to import module 'driver_headers_mbox'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/sphinx/ext/autodoc.py", line 657, in import_object
    __import__(self.modname)
  File "/home/mailinglistparser/mailinglistparser/lib/driver_headers_mbox.py", line 3, in <module>
    from input.check_headers import *
  File "/home/mailinglistparser/mailinglistparser/lib/input/check_headers.py", line 7, in <module>
    from input.imap.header import get_mail_header
  File "/home/mailinglistparser/mailinglistparser/lib/input/imap/header.py", line 7, in <module>
    from util.read_utils import *
  File "/home/mailinglistparser/mailinglistparser/lib/util/read_utils.py", line 6, in <module>
    import networkx as nx
ImportError: No module named 'networkx'
WARNING: /home/mailinglistparser/mailinglistparser/docs/driver_thread_analysis.rst:4: (WARNING/2) autodoc: failed to import module 'driver_thread_analysis'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/sphinx/ext/autodoc.py", line 657, in import_object
    __import__(self.modname)
  File "/home/mailinglistparser/mailinglistparser/lib/driver_thread_analysis.py", line 1, in <module>
    from analysis.author import generate_cl_curve_fits, generate_rt_curve_fits
ImportError: cannot import name 'generate_cl_curve_fits'
WARNING: /home/mailinglistparser/mailinglistparser/docs/input.rst:18: (WARNING/2) autodoc: failed to import module 'input.check_headers'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/sphinx/ext/autodoc.py", line 657, in import_object
    __import__(self.modname)
  File "/home/mailinglistparser/mailinglistparser/lib/input/check_headers.py", line 7, in <module>
    from input.imap.header import get_mail_header
  File "/home/mailinglistparser/mailinglistparser/lib/input/imap/header.py", line 7, in <module>
    from util.read_utils import *
  File "/home/mailinglistparser/mailinglistparser/lib/util/read_utils.py", line 6, in <module>
    import networkx as nx
ImportError: No module named 'networkx'
WARNING: /home/mailinglistparser/mailinglistparser/docs/input.rst:26: (WARNING/2) autodoc: failed to import module 'input.data_cleanup'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/sphinx/ext/autodoc.py", line 657, in import_object
    __import__(self.modname)
  File "/home/mailinglistparser/mailinglistparser/lib/input/data_cleanup.py", line 2, in <module>
    from util.read_utils import lines_per_n
  File "/home/mailinglistparser/mailinglistparser/lib/util/read_utils.py", line 6, in <module>
    import networkx as nx
ImportError: No module named 'networkx'
WARNING: /home/mailinglistparser/mailinglistparser/docs/input.imap.rst:17: (WARNING/2) autodoc: failed to import module 'input.imap.header'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/sphinx/ext/autodoc.py", line 657, in import_object
    __import__(self.modname)
  File "/home/mailinglistparser/mailinglistparser/lib/input/imap/header.py", line 7, in <module>
    from util.read_utils import *
  File "/home/mailinglistparser/mailinglistparser/lib/util/read_utils.py", line 6, in <module>
    import networkx as nx
ImportError: No module named 'networkx'
WARNING: /home/mailinglistparser/mailinglistparser/docs/input.mbox.rst:10: (WARNING/2) autodoc: failed to import module 'input.mbox.keyword_clustering'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/sphinx/ext/autodoc.py", line 657, in import_object
    __import__(self.modname)
  File "/home/mailinglistparser/mailinglistparser/lib/input/mbox/keyword_clustering.py", line 5, in <module>
    from nltk.corpus import stopwords
ImportError: No module named 'nltk'
WARNING: /home/mailinglistparser/mailinglistparser/docs/input.mbox.rst:18: (WARNING/2) autodoc: failed to import module 'input.mbox.keyword_digest'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/sphinx/ext/autodoc.py", line 657, in import_object
    __import__(self.modname)
  File "/home/mailinglistparser/mailinglistparser/lib/input/mbox/keyword_digest.py", line 5, in <module>
    from nltk.corpus import stopwords
ImportError: No module named 'nltk'
WARNING: /home/mailinglistparser/mailinglistparser/docs/input.mbox.rst:26: (WARNING/2) autodoc: failed to import module 'input.mbox.mbox_hdr'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/sphinx/ext/autodoc.py", line 657, in import_object
    __import__(self.modname)
  File "/home/mailinglistparser/mailinglistparser/lib/input/mbox/mbox_hdr.py", line 4, in <module>
    from util.read_utils import *
  File "/home/mailinglistparser/mailinglistparser/lib/util/read_utils.py", line 6, in <module>
    import networkx as nx
ImportError: No module named 'networkx'
WARNING: /home/mailinglistparser/mailinglistparser/docs/test_driver.rst:4: (WARNING/2) autodoc: failed to import module 'test_driver'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/sphinx/ext/autodoc.py", line 657, in import_object
    __import__(self.modname)
  File "/home/mailinglistparser/mailinglistparser/lib/test_driver.py", line 3, in <module>
    from analysis.thread.graph.edge_list import generate_edge_list
  File "/home/mailinglistparser/mailinglistparser/lib/analysis/thread/graph/edge_list.py", line 2, in <module>
    from util.read_utils import lines_per_n
  File "/home/mailinglistparser/mailinglistparser/lib/util/read_utils.py", line 6, in <module>
    import networkx as nx
ImportError: No module named 'networkx'
WARNING: /home/mailinglistparser/mailinglistparser/docs/util.rst:10: (WARNING/2) autodoc: failed to import module 'util.author_uid_map'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/sphinx/ext/autodoc.py", line 657, in import_object
    __import__(self.modname)
  File "/home/mailinglistparser/mailinglistparser/lib/util/author_uid_map.py", line 3, in <module>
    from util.read_utils import lines_per_n
  File "/home/mailinglistparser/mailinglistparser/lib/util/read_utils.py", line 6, in <module>
    import networkx as nx
ImportError: No module named 'networkx'
WARNING: /home/mailinglistparser/mailinglistparser/docs/util.rst:26: (WARNING/2) autodoc: failed to import module 'util.graph_leaf_nodes'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/sphinx/ext/autodoc.py", line 657, in import_object
    __import__(self.modname)
  File "/home/mailinglistparser/mailinglistparser/lib/util/graph_leaf_nodes.py", line 6, in <module>
    from util.read_utils import lines_per_n
  File "/home/mailinglistparser/mailinglistparser/lib/util/read_utils.py", line 6, in <module>
    import networkx as nx
ImportError: No module named 'networkx'
/home/mailinglistparser/mailinglistparser/lib/util/preprocessing_utils.py:docstring of util.preprocessing_utils.is_outlier:9: WARNING: Field list ends without a blank line; unexpected unindent.
WARNING: /home/mailinglistparser/mailinglistparser/docs/util.rst:50: (WARNING/2) autodoc: failed to import module 'util.read_utils'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/sphinx/ext/autodoc.py", line 657, in import_object
    __import__(self.modname)
  File "/home/mailinglistparser/mailinglistparser/lib/util/read_utils.py", line 6, in <module>
    import networkx as nx
ImportError: No module named 'networkx'
WARNING: /home/mailinglistparser/mailinglistparser/docs/util.rst:58: (WARNING/2) autodoc: failed to import module 'util.snaa_curve_fit_layout'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/sphinx/ext/autodoc.py", line 657, in import_object
    __import__(self.modname)
  File "/home/mailinglistparser/mailinglistparser/lib/util/snaa_curve_fit_layout.py", line 2, in <module>
    plt.rcParams['font.size'] = 12
NameError: name 'plt' is not defined
WARNING: /home/mailinglistparser/mailinglistparser/docs/util.rst:66: (WARNING/2) autodoc: failed to import module 'util.thread_graph_nx'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/sphinx/ext/autodoc.py", line 657, in import_object
    __import__(self.modname)
  File "/home/mailinglistparser/mailinglistparser/lib/util/thread_graph_nx.py", line 2, in <module>
    import networkx as nx
ImportError: No module named 'networkx'
looking for now-outdated files... none found
pickling environment... done
checking consistency... /home/mailinglistparser/mailinglistparser/docs/code_documentation.rst: WARNING: document isn't included in any toctree
/home/mailinglistparser/mailinglistparser/docs/driver_author_analysis.rst: WARNING: document isn't included in any toctree
/home/mailinglistparser/mailinglistparser/docs/driver_headers_imap.rst: WARNING: document isn't included in any toctree
/home/mailinglistparser/mailinglistparser/docs/driver_headers_mbox.rst: WARNING: document isn't included in any toctree
/home/mailinglistparser/mailinglistparser/docs/driver_thread_analysis.rst: WARNING: document isn't included in any toctree
/home/mailinglistparser/mailinglistparser/docs/test_driver.rst: WARNING: document isn't included in any toctree
done
preparing documents... done
writing output... [100%] util                                                                                                                  
generating indices... genindex py-modindex
highlighting module code... [100%] util.preprocessing_utils                                                                                    
writing additional pages... search
copying images... [100%] _static/participant_size_table.png                                                                                    
copying static files... done
copying extra files... done
dumping search index in English (code: en) ... done
dumping object inventory... done
build succeeded, 47 warnings.
```
There seems to be quite a few errors and warning. Please check and them and create new issues if needed.

**Comment by yashpungaliya** [comment]
I am aware of the warnings/errors and they arise due to third party imports such as -nltk,sklearn,networkx,etc. 
I don't think they affect the generation of code documentation but I will create an Issue for the same

**Comment by achyudh** [comment]
The script for the documentation has to be run from the correct directory (/etc) which is probably why `ImportError: No module named 'lib'` is coming up. Otherwise, you just have to install the 3rd party libraries like networkx and plotly in the server.

**Comment by achyudh** [comment]
Please try this to  install the 3rd party dependencies:
`"pip install -r requirements.txt"`

**Comment by prasadtalasila** [comment]
@achyudhk I tried your sugestions. Now it says `ImportError: No module named 'sphinx'`; the code I am using is the dev branch 066af2a commit with no modifications.

**Comment by achyudh** [comment]
Can you please check if you are using the correct version of Python? Because the dev branch requirements.txt has `sphinx>=1.4.1`

Please see if both the install of dependencies and running of script are happening through Python3.

**Comment by prasadtalasila** [comment]
With the updated installation setup, the docs are being generated correctly. However, there are still a few helpful warnings.

```python
/home/mailinglistparser/MailingListParser/docs/algos.rst:52: WARNING: Bullet list ends without a blank line; unexpected unindent.
/home/mailinglistparser/MailingListParser/lib/analysis/thread/wh_table.py:docstring of analysis.thread.wh_table.generate_wh_table_threads:6: WARNING: Unexpected indentation.
/home/mailinglistparser/MailingListParser/docs/analysis_techniques.rst:47: WARNING: Inline strong start-string without end-string.
/home/mailinglistparser/MailingListParser/docs/analysis_techniques.rst:204: WARNING: Enumerated list ends without a blank line; unexpected unindent.
/home/mailinglistparser/MailingListParser/lib/input/imap/header.py:docstring of input.imap.header.get_mail_header:1: WARNING: Unknown target name: "range".
/home/mailinglistparser/MailingListParser/lib/input/mbox/keyword_clustering.py:docstring of input.mbox.keyword_clustering.get_message_body:2: WARNING: Field list ends without a blank line; unexpected unindent.
/home/mailinglistparser/MailingListParser/lib/util/preprocess.py:docstring of util.preprocess.is_outlier:7: WARNING: Field list ends without a blank line; unexpected unindent.
looking for now-outdated files... none found
pickling environment... done
checking consistency... /home/mailinglistparser/MailingListParser/docs/code_documentation.rst: WARNING: document isn't included in any toctree
done
```
We can clean up the mistakes in sphinx documentation using the above warnings.

**Comment by hardik124** [comment]
@yashpungaliya  Is this issue still relevant?

**Comment by prasadtalasila** [comment]
Yup. The warnings mentioned in my last message still exist. You can try to rectify the sphinx format errors and remove the warnings.

**Comment by achyudh** [comment]
These warnings are actually due to improper formatting of the reStructuredText comments and have nothing to do with the generation of the documentation.

**Comment by yashpungaliya** [comment]
@prasadtalasila All issues have been resolved and checked on the server! No more warnings appear.

**Comment by prasadtalasila** [comment]
fixed in #80.
